specs/jsonschema-core.md

```diff
@@ -950,40 +949,33 @@ an [absolute IRI](https://www.rfc-editor.org/rfc/rfc3987.html#section-2.2)
-requiring JSON Pointer references to be updated.
-
-The `$anchor` and `$dynamicAnchor` keywords are used to specify such fragments.
-They are identifier keywords that can only be used to create plain name
-fragments, rather than absolute IRIs as seen with `$id`.
-
-`$anchor` defines a reference target for `$ref`. The fragment defined by this
-keyword is appended to the IRI of the schema resource containing it. As
-discussed in {{id-keyword}}, this is either the nearest `$id` in the same or an
-ancestor schema object, or the base IRI for the document as determined according
-to [RFC 3987][rfc3987] and
-[RFC 3986][rfc3986].
-
-In contrast, `$dynamicAnchor` operates independently of resource IRIs and is
-instead dependent on the dynamic scope of the evaluation. `$dynamicAnchor`
-defines a reference target for the `$dynamicRef` keyword. This advanced feature
-makes it easier to extend recursive schemas such as the meta-schemas, without
-imposing any particular semantics on that extension. See {{dynamic-ref}} for
-details.
-
-In most cases, the normal fragment behavior both suffices and is more intuitive.
-Therefore it is RECOMMENDED that `$anchor` be used to create plain name
-fragments unless there is a clear need for `$dynamicAnchor`.
+Using JSON Pointers in IRI fragments to reference subschemas couples the IRI to
+the structure of the schema. Using plain name fragment identifiers in IRI
+fragments to identify subschemas is sometimes preferable because it is not tied
+to a particular structural location. This allows a subschema to be relocated
+without requiring references to be updated.
 
-If present, the value of these keywords MUST be a string and MUST conform to the
-plain name fragment identifier syntax defined in {{fragments}}.[^4]
+The `$anchor` and `$dynamicAnchor` keywords are used to define
+location-independent identifiers for subschemas within a schema resource.
+
+`$anchor` defines a plain name fragment identifier that can be used in IRI
+fragments as an alternative to JSON Pointers.[^4] See {{fragments}}.
 
 [^4]: Note that the anchor string does not include the "#" character, as it is
-not a IRI reference. An `$anchor`: "foo" becomes the fragment `#foo` when used
-in a IRI. See below for full examples.
+just a fragment identifier not an IRI reference. To reference the "foo"
+`$anchor` from the same schema resource, you would use the fragment-only IRI
+`#foo`. See below for full examples.
+
+`$dynamicAnchor` defines a different kind of identifier that only has meaning
+when used as a value for `$dynamicRef`. It's not a fragment identifier that can
+be used in an IRI. `$dynamicRef` uses a syntax similar to a fragment-only IRI,
+but the semantics are different. Unlike fragments in IRIs, the fragment in
+`$dynamicRef` identifiers identifies the primary resource (the schema resource)
+in addition to the secondary resource (the subschema within that schema
+resource). See {{dynamic-ref}} for details.
```

Comment:
This is better.

Maybe highlight that the difference is that the primary resource is being identified with `$dynamicRef`.

Maybe

> Fragments in IRIs can only identify a subschema within the resource identified by the IRI. That is, the fragment itself does not identify the primary schema resource. However, the fragment in `$dynamicRef` identifies both the primary schema resource and a subschema within it. See {{dynamic-ref}} for details.

(Also the "`$dynamicRef` identifiers identifies" is a bit redundant.)

Reply:
I gave this another try. I ended up rewriting most of the paragraph taking a slightly different approach.

> primary schema resource

I specifically wanted to use the terms "primary resource" and "secondary resource" because those are the terms used to define fragment identifiers in RFC-3986. I added a link for people who might not be familiar with those terms.

Reply:
Yeah, your new paragraph is better.  Taking the time to fully explain things and not trying to be terse seems to be the trick here.